tests/unit/CIGitLogicTest.sh

```diff
@@ -127,9 +128,10 @@ function cherry_pick_pr {
   bump_version "$SEMVER_LEVEL_BUILD"
   VERSION_BUMP_COMMIT="$(git rev-parse HEAD)"
 
-  if ! git rev-parse --verify staging 2>/dev/null; then
-    git fetch origin staging --depth=1
-  fi
+  checkout_repo
```

Comment:
This was the line that was missing to make this test more like "the real world" – it checks out the repo as actions/checkout does and therefore resets the client repo to have only 1 commit of history on the main branch.

If you want to see the test fail + reproduce the same error we saw in production, change line 134 to:

```sh
git fetch origin staging "$VERSION_BUMP_COMMIT" "$PR_MERGE_COMMIT" --no-tags --depth=1
```

Then you'll see an incorrect conflict in package.json and package-lock.json.